### PR TITLE
code refactoring

### PR DESCRIPTION
**Pull request by gurbinder533**
There are lots of big functions that need to be broken into smaller pieces. I have started doing it but need more time. Please keep reviewing this to make sure I am not changing anything fundamental.

### CODE REVIEW

File: lonestar/eda/cpu/bipart-nested/Bipart.cpp [hunk 1]
```diff
@@ -352,60 +335,60 @@ int main(int argc, char** argv) {
...
-  galois::preAlloc(galois::runtime::numPagePoolAllocTotal() * 20);
-  galois::reportPageAlloc("MeminfoPre");
+void CreatePartitions(MetisGraph& metis_graph) {
+  GGraph& graph            = *metis_graph.getGraph();
+  uint64_t total_num_nodes = graph.size();
+  uint64_t num_hedges      = graph.hedges;
   //! FIXME(HCLee) why is it vector?
   std::vector<MetisGraph*> vecGraph;
-  vecGraph.push_back(&metisGraph);
+  vecGraph.push_back(&metis_graph);
```
Comment (nicelhc13) [inline lonestar/eda/cpu/bipart-nested/Bipart.cpp:349]: let's replace it with vecGraph --> graph_vec or graph_lists

File: lonestar/eda/cpu/bipart-nested/Bipart.cpp [hunk 2]
```diff
@@ -352,60 +335,60 @@ int main(int argc, char** argv) {
...
+  std::memset(value_array, 0, num_partitions * sizeof(uint32_t));
+  std::vector<uint32_t> vec(num_partitions);
+
+  uint32_t second_partition     = (num_partitions + 1) / 2;
+  value_array[0]                = second_partition;
+  value_array[second_partition] = num_partitions / 2;
+
+  galois::do_all(galois::iterate(num_hedges, total_num_nodes), [&](uint64_t n) {
+    MetisNode& node   = graph.getData(n);
+    uint32_t nodePart = node.getPart();
```
Comment (nicelhc13) [inline lonestar/eda/cpu/bipart-nested/Bipart.cpp:369]: nodePart --> part_of_node

File: lonestar/eda/cpu/bipart-nested/Bipart.cpp [hunk 3]
```diff
@@ -479,51 +462,51 @@ int main(int argc, char** argv) {
     ///< Tmp log 3): graph indices of the hedges are assigned.
 
     //! The currently processed number of partitions.
-    int len = toProcess.size();
+    int len = current_level_indices.size();
```
Comment (nicelhc13) [inline lonestar/eda/cpu/bipart-nested/Bipart.cpp:465]: int len --> uint32_t len

File: lonestar/eda/cpu/bipart-nested/Bipart.cpp (hunk 1 above)
Comment (gurbinder533) [reply lonestar/eda/cpu/bipart-nested/Bipart.cpp]: I am changing this to "metis_graphs".

**Pull request merged**